qiskit/algorithms/time_evolvers/trotterization/trotter_qrte.py

```diff
@@ -0,0 +1,160 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""An algorithm to implement a Trotterization real time-evolution."""
+
+from __future__ import annotations
+
+from qiskit.algorithms.observables_evaluator import eval_observables
+
+from qiskit import QuantumCircuit
+from qiskit.algorithms.time_evolvers.evolution_problem import EvolutionProblem
+from qiskit.algorithms.time_evolvers.evolution_result import EvolutionResult
+from qiskit.algorithms.time_evolvers.real_evolver import RealEvolver
+from qiskit.opflow import (
+    CircuitOp,
+    PauliSumOp,
+    StateFn,
+)
+from qiskit.circuit.library import PauliEvolutionGate
+from qiskit.primitives import BaseEstimator
+from qiskit.quantum_info import Pauli
+from qiskit.synthesis import ProductFormula, LieTrotter
+
+
+class TrotterQRTE(RealEvolver):
+    """Quantum Real Time Evolution using Trotterization.
+    Type of Trotterization is defined by a ProductFormula provided.
+
+    Examples:
+
+        .. jupyter-execute::
```

Comment:
This is out of curiosity, what does this "jupiter-execute" line do?

Reply:
Not sure actually, changed to `.. code-block:: python` as other algorithms!

Reply:
FYI https://jupyter-sphinx.readthedocs.io/en/latest/

As a code-block this will not be executed and if things change any failure to update the sample here accordingly may render it out of date and fails to work. I understand qiskit is moving away from jupyter-execute as its kinda expensive to run. [Doctest ](https://docs.python.org/3/library/doctest.html)is an option that has come up - which is used in the app repos to run/test code in these blocks.